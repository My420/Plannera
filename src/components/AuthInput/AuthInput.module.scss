@import '../../sass/variables.scss';

.label {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  width: 300px;
}

.input {
  box-sizing: border-box;

  width: 300px;
  height: 40px;

  padding: 0 5px;

  border: none;
  border-radius: 3px;

  font-size: 14px;
  line-height: 16px;
  text-align: center;
  color: $fonts-auth-color;
}

.error,
.valid {
  min-height: 14px;

  padding: 10px 2px 0 2px;
  font-size: 12px;
  line-height: 14px;
  color: $fonts-app-error-color;
}

.valid {
  opacity: 0;
}

.notValid {
  background-color: red;
}
