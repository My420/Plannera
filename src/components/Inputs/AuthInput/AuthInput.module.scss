@import '../../../sass/variables.scss';
@import '../../../sass/mixins.scss';

.label {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  width: 300px;
}

.input {
  @include auth-input;
  box-sizing: border-box;

  width: 300px;
  height: 40px;

  padding: 0 5px;

  border: none;
  border-radius: 3px;

  font-size: 14px;
  line-height: 16px;
  text-align: center;
  color: $fonts-auth-color;
}

.error,
.valid {
  min-height: 14px;

  padding: 10px 2px 0 2px;
  font-size: 12px;
  font-weight: 700;
  line-height: 14px;
  color: $fonts-app-error-color;
}

.valid {
  opacity: 0;
}

.notValid {
  background-color: red;
}

@media (max-width: $mobile-large) {
  .label {
    width: 280px;
  }

  .input {
    width: 280px;
    height: 30px;
  }

  .error,
  .valid {
    min-height: 11px;
    font-size: 11px;
    line-height: 11px;
  }
}
