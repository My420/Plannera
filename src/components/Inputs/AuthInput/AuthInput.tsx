import React from 'react';
import { InputAttributeType } from '../../../types/types';
import styles from './AuthInput.module.scss';

export interface AuthInputProps extends React.HTMLAttributes<HTMLInputElement> {
  label: string;
  type: InputAttributeType;
  name: string;
  id: string;
  placeholder: string;
  extensionContainerClass?: string;
  extensionInputClass?: string;
  value: string;
  isValid: boolean;
  error: string;
  onInputChange: (name: string, value: string) => void;
}

const AuthInput: React.FC<AuthInputProps> = ({
  value,
  onInputChange,
  label,
  name,
  id,
  type,
  placeholder,
  extensionContainerClass,
  extensionInputClass,
  isValid,
  error,
  ...rest
}) => {
  const handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
    const { name: inputName, value: newValue } = evt.currentTarget;
    onInputChange(inputName, newValue);
  };

  return (
    <label htmlFor={id} className={`${styles.label} ${extensionContainerClass}`}>
      <span className="visually-hidden">{label}</span>
      <span className={`${isValid ? styles.valid : styles.error}`}>
        {error}
        {' '}
      </span>
      <input
        value={value}
        id={id}
        className={`${styles.input} ${extensionInputClass}`}
        name={name}
        type={type}
        placeholder={placeholder}
        onChange={handleChange}
        {...rest}
      />
    </label>
  );
};

AuthInput.defaultProps = {
  extensionContainerClass: '',
  extensionInputClass: '',
};

export default AuthInput;
