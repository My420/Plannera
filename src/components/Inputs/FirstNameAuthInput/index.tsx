import FirstNameAuthInput from './FirstNameAuthInput';

export default FirstNameAuthInput;
