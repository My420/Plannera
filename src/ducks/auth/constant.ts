import { fb as firebase } from '../../firebase/config';

export const moduleName = 'auth';
export const SIGN_UP_REQUEST = 'Plannera/auth/SIGN_UP_REQUEST';
export const SIGN_UP_SUCCESS = 'Plannera/auth/SIGN_UP_SUCCESS';
export const SIGN_UP_ERROR = 'Plannera/auth/SIGN_UP_ERROR';
export const SIGN_IN_REQUEST = 'Plannera/auth/SIGN_IN_REQUEST';
export const SIGN_IN_SUCCESS = 'Plannera/auth/SIGN_IN_SUCCESS';
export const SIGN_IN_ERROR = 'Plannera/auth/SIGN_IN_ERROR';
export const SIGN_OUT_REQUEST = 'Plannera/auth/SIGN_OUT_REQUEST';
export const SIGN_OUT_SUCCESS = 'Plannera/auth/SIGN_OUT_SUCCESS';
export const SIGN_OUT_ERROR = 'Plannera/auth/SIGN_OUT_ERROR';
export const AUTH_CLEAR_ERROR = 'Plannera/auth/AUTH_CLEAR_ERROR';

export const auth = firebase.auth();
