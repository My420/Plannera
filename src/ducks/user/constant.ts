export const moduleName = 'user';
export const CREATE_USER_REQUEST = 'Plannera/user/CREATE_USER_REQUEST';
export const CREATE_USER_SUCCESS = 'Plannera/user/CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'Plannera/user/CREATE_USER_ERROR';

export const LOAD_USER_REQUEST = 'Plannera/user/LOAD_USER_REQUEST';
export const LOAD_USER_SUCCESS = 'Plannera/user/LOAD_USER_SUCCESS';
export const LOAD_USER_ERROR = 'Plannera/user/LOAD_USER_ERROR';

export const CHANGE_USER_DATA_REQUEST = 'Plannera/user/CHANGE_USER_DATA_REQUEST';
export const CHANGE_USER_DATA_SUCCESS = 'Plannera/user/CHANGE_USER_DATA_SUCCESS';
export const CHANGE_USER_DATA_ERROR = 'Plannera/user/CHANGE_USER_DATA_ERROR';

export const CLEAR_USER_STATE = 'Plannera/user/CLEAR_USER_STATE';
