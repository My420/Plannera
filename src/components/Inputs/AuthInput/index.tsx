import React from 'react';
import AuthInput, { AuthInputProps } from './AuthInput';

const areEqual = (prevProps: AuthInputProps, nextProps: AuthInputProps) => {
  const { value: prevValue, isValid: prevIsValid, error: prevError } = prevProps;
  const { value: nextValue, isValid: nextIsValid, error: nextError } = nextProps;
  return prevValue === nextValue && prevIsValid === nextIsValid && prevError === nextError;
};

export default React.memo(AuthInput, areEqual);
