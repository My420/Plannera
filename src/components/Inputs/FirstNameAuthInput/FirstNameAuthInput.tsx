import React from 'react';
import { FIRST_NAME } from '../../../utils/constant';
import AuthInput from '../AuthInput/AuthInput';

export interface FirstNameAuthInputProps {
  extensionContainerClass?: string;
  extensionInputClass?: string;
  value: string;
  isValid: boolean;
  error: string;
  onInputChange: (name: string, value: string) => void;
}

const FirstNameAuthInput: React.FC<FirstNameAuthInputProps> = (props) => (
  <AuthInput
    id="firstName"
    name={`${FIRST_NAME}`}
    type="text"
    placeholder="First name"
    label="First name"
    {...props}
  />
);

FirstNameAuthInput.defaultProps = {
  extensionContainerClass: '',
  extensionInputClass: '',
};

export default FirstNameAuthInput;
