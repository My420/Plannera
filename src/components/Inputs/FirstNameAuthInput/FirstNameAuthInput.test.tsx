import React from 'react';
import { shallow } from 'enzyme';
import { AuthInputProps } from '../AuthInput/AuthInput';
import { FIRST_NAME } from '../../../utils/constant';
import FirstNameAuthInput, { FirstNameAuthInputProps } from './FirstNameAuthInput';

describe('test FirstNameAuthInput component', () => {
  const props: FirstNameAuthInputProps = {
    value: '',
    isValid: false,
    error: '',
    onInputChange: () => {},
    extensionContainerClass: 'testContainerClass',
    extensionInputClass: 'testInputClass',
  };
  const wrapper = shallow(<FirstNameAuthInput {...props} />);

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render Memo(AuthInput) component with expected props', () => {
    const expectedProps: AuthInputProps = {
      id: 'firstName',
      name: FIRST_NAME,
      type: 'text',
      placeholder: 'First name',
      label: 'First name',
      ...props,
    };
    expect(wrapper.find('Memo(AuthInput)').props()).toEqual(expectedProps);
  });
});
