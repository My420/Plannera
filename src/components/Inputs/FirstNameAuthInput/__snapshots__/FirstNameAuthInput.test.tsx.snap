// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test FirstNameAuthInput component should match snapshot 1`] = `
<Memo(AuthInput)
  error=""
  extensionContainerClass="testContainerClass"
  extensionInputClass="testInputClass"
  id="firstName"
  isValid={false}
  label="First name"
  name="FIRST_NAME"
  onInputChange={[Function]}
  placeholder="First name"
  type="text"
  value=""
/>
`;
